pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go

```diff
@@ -150,6 +153,104 @@ func TestFindAndAddNewPods_WithRescontructedVolume(t *testing.T) {
 
 }
 
+func TestFindAndAddNewPods_WithDifferentConditions(t *testing.T) {
+	tests := []struct {
+		desc           string
+		hasAddedPods   bool
+		expectedFound  bool // Found pod is added to DSW
```

Comment:
To organize the arguments better, I think putting the expected result at the end would make it easier to read.

Reply:
Modified